src/rgw/rgw_url.cc

```diff
@@ -13,7 +13,7 @@ namespace {
 
   const std::string schema_re = "([[:alpha:]]+:\\/\\/)";
   const std::string user_pass_re = "(([^:\\s]+):([^@\\s]+)@)?";
-  const std::string host_port_re = "([[:alnum:].:-]+)";
+  const std::string host_port_re = "([[:alnum:],.:-]+)";
```

Comment:
valid urls don't support comma-separated hostnames

Reply:
Yes, kafka actually handle urls, you can read from [here](https://github.com/confluentinc/librdkafka/blob/a5c5641c9efff1d09993098fc8ed1b8935d2b86a/examples/consumer.c#L107)

When you work with a kafka cluster, the most easiest way is to setup connection string with a comma seperated of host:port.

Reply:
you can see some example [here](https://github.com/confluentinc/librdkafka/issues/943)

Reply:
we need this fix, but agree with @cbodley we should not change the url validation code.
lets add a broker list as an additional attribute to the topic creation.
see: https://github.com/ceph/ceph/blob/main/src/rgw/driver/rados/rgw_pubsub_push.cc#L306
we should continue to support the existing url method, for backward compatibility, and have this attribute as "broker-list", which would be a comma separated list of URLs.

please make sure that your change is compatible with: https://github.com/ceph/ceph/pull/56493
should we allow only one user/password for all brokers?
if so, we should probably prevent providing user/password in the URLs of the broker list

Reply:
As I know, multiple brokers (in one kafka cluster) share username and password, so we can use kafka://<username>:<password>@brk1:port1,brk2:port2

I've registerd account in tracker. I will create issue when account is activated.

Reply:
> As I know, multiple brokers (in one kafka cluster) share username and password, so we can use kafka://:@brk1:port1,brk2:port2
> 

ok. so my suggestion here is:
* keep the `push-endpoint` parameter without a change. this has to remain there for backward cmpatibility. if we need more than one broker, we must still use this parameter, as this would give us an indication that we have a kafka endpoint. and also, in the code that would identify the "connection"
* when we want to pass a broker list, we would use a new parameter `broker-list` that would have comma separated list of `host:port` (without any other URL parts)
* user/password would be taken from the `push-endpoint` and once the above PR is merged, it would be taken from the new `user-name` and `password` parameters
* to simplify things, we should probably allow overlap between `push-endpoint` and the new `broker-list` parameter

Reply:
can we overwrite the endpoint from broker-list param? (endpoint will be formed with kafka://username:password@brk1:port1,brk2:port2

Reply:
Sounds good